pages/articles/ArticlesHandler.inc.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/**
+ * @file pages/articles/ArticlesHandler.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ArticlesHandler
+ * @ingroup pages_articles
+ *
+ * @brief Handle requests for articles archive functions.
+ */
+
+import('classes.handler.Handler');
+
+class ArticlesHandler extends Handler
```

Comment:
I didn't specify it in my original post, but the intention is for all articles to be available under the same route too. So instead of:

```
/article/view/1
/articles/section/reviews
```

We would have:

```
/articles/view/1
/articles/section/reviews
```

The idea is that everything is nested under the `/<page>/` directory.

Reply:
We would mimic that in OMP with:

```
/catalog/view/1
/catalog/series/reviews
```

Reply:
I was already going to ask what is the difference here but not I can see the additional S letter there :)

Yes sure, I guess the paths were so close that I just missed that.

Reply:
In OMP you probably mean:
```
/catalog/book/1
/catalog/series/reviews
```

Reply:
Hey so that means moving stuff from article/ArticleHandler to ArticlesHandler

Do we want to rename also the hooks there, like `ArticleHandler::view`

Reply:
You can combine them in one handler or keep them separate like this:

```php
switch ($op) {
    // https://.../articles/view/1
    case 'view':
    case 'download':
        define('HANDLER_CLASS', 'APP\pages\article\ArticleHandler');
        break;
    // https://.../articles/sections
    case 'section':
        define('HANDLER_CLASS', 'APP\pages\article\SectionsHandler');
        break;
    // https://.../articles/
    default:
        define('HANDLER_CLASS', 'APP\pages\article\ArticlesHandler');
}
```

It's up to you. But yeah I'd probably keep all of the `ArticleHandler::*****` hooks around for backwards compatibility where possible.